site/src/components/TemplateParameters/TemplateParameters.tsx

```diff
@@ -5,9 +5,11 @@ import {
   RichParameterInputProps,
 } from "components/RichParameterInput/RichParameterInput";
 import { ComponentProps, FC } from "react";
+import { AutofillSource } from "utils/richParameters";
 
 export type TemplateParametersSectionProps = {
   templateParameters: TemplateVersionParameter[];
+  autofillSources?: Record<string, AutofillSource>;
```

Comment:
```suggestion
  autofillSources?: Record<string, AutofillSource?>;
```
I imagine this change is more truthful, unless Records somehow override this (been a while since I wrote TS types).

Reply:
Hmm, there is already the concept of non-existence built into the record type, and I don't the type to allow for an autofill without a specified source.